grout/views.py

```diff
@@ -47,13 +55,85 @@ def get_serializer_class(self):
 
 
 class RecordViewSet(viewsets.ModelViewSet):
-    queryset = Record.objects.all()
     serializer_class = RecordSerializer
     filter_class = RecordFilter
     bbox_filter_field = 'geom'
     jsonb_filter_field = 'data'
-    filter_backends = (InBBoxFilter, JsonBFilterBackend,
-                       DjangoFilterBackend, DateRangeFilterBackend)
+    filter_backends = (InBBoxFilter, JsonBFilterBackend, DateRangeFilterBackend)
+
+    VALID_GEOTYPES = ['point', 'polygon', 'none']
+    GEOTYPE_NOT_VALID = "The 'geometry_type' parameter must be one of: " + ', '.join(VALID_GEOTYPES)
+    GEOTYPE_AND_POLYGON = ("The 'geometry_type' parameter cannot be 'none' when " +
+                           "either of the 'polygon' or 'polygon_id' parameters is present.")
+    INVALID_DATETIMES = "'{none}' cannot be 'none' when '{timestamp}' is a timestamp."
+
+    def get_queryset(self):
+        '''
+        Check for the `geometry_type` and `occurred_min`/`occurred_max`
+        query parameters. If they exist, use them to generate a queryset for
+        the appropriate class of Record.
```

Comment:
That's an interesting idea! For clarity, when you suggest moving geometry type and datetime information up to `RecordType`, are you thinking something like this:

```
RecordType(GroutModel):
    geometry_type = models.ChoiceField()
    temporal = models.BooleanField(default=True)
```

Points 1 and 2 above seem reasonable to me, although I'm on the fence about 3. I think the fliers app provides an interesting example edge case: for this use-case, polygons are attractive because sometimes you don't know the exact address of an event (or the event organizers don't want it to be made public), so the archivists would want to store a more general boundary instead of a point. In this case, it feels most intuitive for the two cases (a specific address point vs. a general boundary) to be the same type of Record, since they're attempting to catalog the same type of entities in the world. But I could imagine an implementation that separates them; you'd just have to have two record types, e.g. `Event` and `BoundaryEvent`, and make sure to query for both when appropriate. Plus, from a code perspective it might end up being more tedious to receive `Record`s with mixed geometries from the same API query, since then you have to do an additional filter to figure out how to display them.

All that being said, at this point I think the benefits of your proposal outweigh the costs. I can't think of any kind of query functionality that is made impossible by this change -- it just seems like it makes certain operations (querying for all `Record`s regardless of `RecordType`, querying for multiple geometry types) more verbose, but there's still uncertainty in whether those operations would be attractive to most users anyway.

Reply:
That's an interesting use-case! So the idea is either to a) convey uncertainty about a location or b) to obscure the true location? I agree that storing these types of records as mixed geometries would be a bit of a pain to handle on the frontend. 

I wonder if there might be other ways to get at the same functionality that would keep filtering easy. For example, would it work to add a checkbox to the data entry form to obscure the true location? If that box was checked, the frontend could jitter the entered point by a certain amount, say 100 meters, before POSTing to the API. For uncertain location, perhaps the schema could have a toggle or numeric entry for uncertainty that would cause the frontend to draw a bubble rather than a point for that record if it was checked?

I think that handling uncertainty and obscuring true locations (potentially with the option to display it to administrators) are a common enough need that it might be helpful to add functionality to help with this to Grout. I've definitely encountered apps that have to grapple with one or more of these things before, but I haven't put much thought into how Grout might make it easier to handle these types of issues specifically.